upup/models/nodeup/docker/_systemd/_debian_family/files/opt/kubernetes/helpers/docker-healthcheck

```diff
@@ -26,20 +26,26 @@ echo "docker failed"
 echo "Giving docker 30 seconds grace before restarting"
 sleep 30
 
-if timeout 10 docker ps > /dev/null; then
+if timeout 60 docker ps > /dev/null; then
   echo "docker recovered"
   exit 0
 fi
 
-echo "docker still down; triggering docker restart"
-systemctl restart docker
+echo "docker still unresposive; triggering docker restart"
+systemctl stop docker
 
-echo "Waiting 60 seconds to give docker time to start"
+echo "wait all tcp sockets to close"
+sleep `cat /proc/sys/net/ipv4/tcp_fin_timeout`
+
+sleep 10
+systemctl start docker
+
+echo "Waiting 120 seconds to give docker time to start"
```

Comment:
Remove the `sleep 60` and update the if statement to `if timeout 120 docker ps > /dev/null; then` for readability

Reply:
I though the same thing.  But maybe the sleep here because `timeout` doesn't retry, so we need to wait for docker to have the change to start listening cc @tatobi ?